pkg/v1/cli/artifact/oci.go

```diff
@@ -17,6 +26,27 @@ func NewOCIArtifact(image string) Artifact {
 
 // Fetch an artifact.
 func (g *OCIArtifact) Fetch() ([]byte, error) {
-	// TODO(anujc25): implement OCI artifact fetch
-	return nil, nil
+	filesMap, err := carvelhelpers.GetFilesMapFromImage(g.Image)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable fetch plugin binary")
+	}
+
+	var bytesData []byte
+	fileCount := 0
+
+	for path, fileData := range filesMap {
+		// Skip any testing related directory paths if bundled
+		if utils.ContainsString(filepath.SplitList(path), "test") {
```

Comment:
Is there some convention where these test related directories are often bundled in the image that we have to explicitly filter them? If so are they always named 'test'?

Reply:
The publish script takes the artifacts directory as an input. And for publishing the artifacts, It traverse to the specific plugin/version artifact directory at bundles the plugins from there. Sometimes, if the `test` plugin directory exists the bundle also includes `test` plugin directory, and this change just trying to skip that if it exists.

To improve around this, we can make some changes in the publish script to make sure it only publishes plugin binary and skip the publishing of the directory. But this check has been added as a safeguard for now.